code/datums/components/storage/concrete/pockets.dm

```diff
@@ -17,16 +17,30 @@
 
 /datum/component/storage/concrete/pockets/small
 	max_items = 1
+	max_w_class = WEIGHT_CLASS_SMALL
 	attack_hand_interact = FALSE
 
 /datum/component/storage/concrete/pockets/tiny
 	max_items = 1
 	max_w_class = WEIGHT_CLASS_TINY
 	attack_hand_interact = FALSE
 
-/datum/component/storage/concrete/pockets/small/detective
+/datum/component/storage/concrete/pockets/small/detective/Initialize()
+	. = ..()
 	attack_hand_interact = TRUE // so the detectives would discover pockets in their hats

```

Comment:
Why did you move this into initialize?

Reply:
Needed initialize to get access to exception_hold

Reply:
no why was `attack_hand_interact` moved to initialize

Reply:
Changed this now